### PR TITLE
Regression: fix `directory` endpoint not listing teams

### PR DESCRIPTION
**Pull request by carlosrodrigues94**

### CODE REVIEW

File: apps/meteor/server/methods/browseChannels.js [hunk 1]
```diff
@@ -143,19 +149,23 @@ async function getTeams(user, searchTerm, sort, pagination) {
...
+
+	const teams = [];
+
+	for await (const room of rooms) {
+		const result = await getChannelsCountForTeam(room.teamId).toArray();
+
+		teams.push({ ...room, roomsCount: result.length });
+	}
+
+	const [total] = await Promise.all([totalCount]);
```
Comment (matheusbsilva137) [inline apps/meteor/server/methods/browseChannels.js:164]:
```suggestion
	const [rooms, total] = await Promise.all([cursor.toArray(), totalCount]);
	const teams = [];

	for (const room of rooms) {
		const roomsCount = await getChannelsCountForTeam(room.teamId).count();
		teams.push({ ...room, roomsCount });
	}
```
I think we can use `Promise.all` here to get all the data at once. Also, I think we don't need to use a `for await` (I've tested it and it looks like just a `for` should be enough here).

File: apps/meteor/server/methods/browseChannels.js [hunk 2]
```diff
@@ -104,16 +104,22 @@ async function getChannelsAndGroups(user, canViewAnon, searchTerm, sort, paginat
...
+	},
+);
 
 async function getTeams(user, searchTerm, sort, pagination) {
 	if (!user) {
 		return;
 	}
 
 	const userSubs = Subscriptions.cachedFindByUserId(user._id).fetch();
+
```
Comment (matheusbsilva137) [inline apps/meteor/server/methods/browseChannels.js:122]:
```suggestion
```
just erasing this empty line

**Review by matheusbsilva137 — CHANGES_REQUESTED** [review]

File: apps/meteor/server/methods/browseChannels.js [hunk 3]
```diff
@@ -143,19 +149,19 @@ async function getTeams(user, searchTerm, sort, pagination) {
...
-		totalCount,
-	]);
+	const [rooms, total] = await Promise.all([cursor.toArray(), totalCount]);
+
+	const teams = [];
+
+	for await (const room of rooms) {
+		const result = await getChannelsCountForTeam(room.teamId).count();
+
+		teams.push({ ...room, roomsCount: result.length });
```
Comment (matheusbsilva137) [inline apps/meteor/server/methods/browseChannels.js:159]:
```suggestion
		const roomsCount = await getChannelsCountForTeam(room.teamId).count();
		teams.push({ ...room, roomsCount });
```

**Review by matheusbsilva137 — CHANGES_REQUESTED** [review]

**Review by matheusbsilva137 — APPROVED** [review]

File: apps/meteor/server/methods/browseChannels.js [hunk 4]
```diff
@@ -104,16 +104,22 @@ async function getChannelsAndGroups(user, canViewAnon, searchTerm, sort, paginat
 	};
 }
 
-const getChannelsCountForTeam = mem((teamId) => Promise.await(Rooms.findByTeamId(teamId, { projection: { _id: 1 } }).count()), {
-	maxAge: 2000,
-});
+const getChannelsCountForTeam = mem(
+	(teamId) => {
+		return Rooms.findByTeamId(teamId, { projection: { _id: 1 } });
```
Comment (ggazzo) [inline apps/meteor/server/methods/browseChannels.js:109]: as far as i understand this returns a cursor, cursors are VERY FAST, they only  generates a object structure, no cache needed

**Review by sampaiodiego — APPROVED** [review]

**Comment by sampaiodiego** [comment]
that endpoint is missing a test to validate the return of teams.. can either of you @carlosrodrigues94 or @matheusbsilva137 add a test to it pls?

**Pull request merged**